### PR TITLE
Resolves Circular Dependency issue

### LINKED ISSUE #797: Circular reference error when autowiring JobBuilderFactory in task applications
As discussed in https://github.com/spring-projects/spring-batch/issues/3991 if a user includes a bean required for a `JobBuilderFactory` a BeanCurrentlyInCreationException.   This also affects `JdbcCursorItemReaderAutoConfiguration` and `FlatFileItemReaderAutoConfiguration`.

### PR DESCRIPTION
**Pull request by cppwfs**
Makes sure that dependencies are passed in via parameters in bean declaraction vs. Autowiring

refer to https://github.com/spring-projects/spring-batch/issues/3991
resolves #797

### CODE REVIEW

File: spring-cloud-starter-single-step-batch-job/src/main/java/org/springframework/cloud/task/batch/autoconfigure/flatfile/FlatFileItemReaderAutoConfiguration.java [hunk 1]
```diff
@@ -84,24 +75,24 @@ public FlatFileItemReaderAutoConfiguration(FlatFileItemReaderProperties properti
 				.comments(this.properties.getComments()
 						.toArray(new String[this.properties.getComments().size()]));
 
-		if (this.lineTokenizer != null) {
-			mapFlatFileItemReaderBuilder.lineTokenizer(this.lineTokenizer);
+		if (lineTokenizer != null) {
+			mapFlatFileItemReaderBuilder.lineTokenizer(lineTokenizer);
```
Comment (mminella) [inline spring-cloud-starter-single-step-batch-job/src/main/java/org/springframework/cloud/task/batch/autoconfigure/flatfile/FlatFileItemReaderAutoConfiguration.java:79]: Why add the if statements? We previously depended upon the builder to validate that the state was valid.

Comment (cppwfs) [reply spring-cloud-starter-single-step-batch-job/src/main/java/org/springframework/cloud/task/batch/autoconfigure/flatfile/FlatFileItemReaderAutoConfiguration.java:79]: They were not required so , they can be null.   So we need the null check.

Comment (mminella) [reply spring-cloud-starter-single-step-batch-job/src/main/java/org/springframework/cloud/task/batch/autoconfigure/flatfile/FlatFileItemReaderAutoConfiguration.java:79]: Again...the builder does that check. We don't do null checks on any of the other places we do this kind of thing (see [here](https://github.com/spring-cloud/spring-cloud-task/pull/798/files#diff-af8f3bbf8f60a5583ee99cc0d35b7938d5838ccdcb4f8361a979fb4e16e96748R75) for example).

Comment (cppwfs) [reply spring-cloud-starter-single-step-batch-job/src/main/java/org/springframework/cloud/task/batch/autoconfigure/flatfile/FlatFileItemReaderAutoConfiguration.java:79]: Bye bye `if`s

**Pull request merged**